modules/scala-3-step-filter/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilterBridge.scala

```diff
@@ -110,11 +119,30 @@ class ScalaStepFilterBridge(
           case _ => { false }
       }
       case _ => false
+  def isAndOrFunction(tpe: Type): Boolean =
+    tpe match
+      case ref: TypeRef => {
+        ref.prefix match
+          case t: PackageRef =>
+            ((ref.name.toString.equals("|") || ref.name.toString
+              .equals("&")) & t.fullyQualifiedName.toString().equals("scala"))
```

Comment:
In Scala we prefer to use `==`.

```suggestion
            (ref.name.toString == "|" || ref.name.toString == "&") &&
              t.fullyQualifiedName.toString == "scala"
```